apps/admin-gui/src/app/admin/pages/admin-page/admin-services/service-detail-page/service-required-attributes/service-required-attributes.component.html

```diff
@@ -0,0 +1,39 @@
+<div>
+  <h1 class="page-subtitle">{{ 'SERVICE_DETAIL.REQUIRED_ATTRIBUTES.TITLE' | translate}}</h1>
+  <perun-web-apps-refresh-button
+    *ngIf="!loading"
+    (refresh)="refreshTable()">
+  </perun-web-apps-refresh-button>
+  <button
+    *ngIf="!loading"
+    mat-flat-button
+    color="accent"
+    class="ml-2"
+    (click)="onAdd()">
+    {{ 'SERVICE_DETAIL.REQUIRED_ATTRIBUTES.ADD' | translate}}
+  </button>
+  <button
+    *ngIf="!loading"
+    mat-flat-button
+    color="warn"
+    class="ml-2"
+    [disabled]="selection.selected.length === 0"
+    (click)="onRemove()">
+    {{ 'SERVICE_DETAIL.REQUIRED_ATTRIBUTES.REMOVE' | translate}}
+  </button>
+  <perun-web-apps-immediate-filter
+    (filter)="applyFilter($event)"
+    *ngIf="!loading"
+    [placeholder]="'SERVICE_DETAIL.REQUIRED_ATTRIBUTES.FILTER'">
+  </perun-web-apps-immediate-filter>
+  <mat-spinner class="ml-auto mr-auto" *ngIf="loading"></mat-spinner>
+  <app-attr-def-list
+    *ngIf="!loading"
+    [pageSize]="pageSize"
+    (page)="pageChanged($event)"
+    [definitions]="attrDefinitions"
+    [selection]="selection"
+    [filterValue]="filterValue"
+    (refreshEvent)="refreshTable()">
+  </app-attr-def-list>
```

Comment:
when you click on the row of the list the edit dialog for the attribute is shown, which is not bad behaviour, just its bad when in that dialog it is list of services that are clickable - I would make them unclickable.(it looks really weird when you click on them)